code/common/cloudutils.q

```diff
@@ -0,0 +1,59 @@
+/a function to execute system commands and return a log message depending on the resulting exit code:
+/used for both launch and killproc
+syscomm:{[params;cmd]
+    /params is (i) a dictionary if syscomm has been called by launch, and (ii) a single string if it has been called by killproc
+    /cmd is the command to be executed as a string
+    
+    prevexitcode:first "I"$system cmd,"; echo $?";
+ 
+    $[99h=type params;
+        [$[0=prevexitcode;
+            .lg.o[`launchprocess;"Successful execution: Starting ",params[`procname]];
+          1=prevexitcode;
+            .lg.e[`launchprocess;"Error: ",params[`procname]," failed to start"];
+          2=prevexitcode;
+            .lg.e[`launchprocess;"Error: ",params[`procname]," already running"];
+          .lg.e[`launchprocess;"Unknown error encountered"]]];
+        
+        [$[0=prevexitcode;
+            .lg.o[`killprocess;"Successful execution: Terminating ",params];
+          1=prevexitcode;
+            .lg.e[`killprocess; params," was not running"];
+          2=prevexitcode;
+            .lg.e[`killprocess;"Process name ",params," was not found"];
+          3=prevexitcode;
+            .lg.e[`killprocess;"Failed to terminate "params]
+          .lg.e[`killprocess;"Unknown error encountered"]]] 
+      ]
+   }
+
+/function which lets us call launchprocess.sh from inside a TorQ process
+/it takes a dictionary of parameters which will be passed to launchprocess.sh, i.e "-procname rdb1 -proctype rdb" etc.
+launch:{ [params]
+    /exit immediately if process name and type aren't provided
+    if[not 11b~`procname`proctype in key params;0N!
+        .lg.e[`launchprocess;"Process name and type must be provided"];
+        :()];
+
+    /set default values with .Q.def and string the result:
+    deflt:`procname`proctype`U`localtime!(`;`;`$getenv[`KDBAPPCONFIG],"/passwords/accesslist.txt";1);
```

Comment:
I think default for localtime should be `.proc.localtime` i.e. whatever it's set to in the parent process.

I think we should possibly also try to use the `-U` setting from the parent process, maybe falling back to what you have here if that's not present.

I think we can do something like:

```
first`$.Q.opt[.z.X]`U
```

to try and get the `-U` arg of the current process (`.proc.params` that is defined in TorQ uses `.z.x` rather than `.z.X`, so it doesn't include kdb's single letter args like `-U`)

I'd consider inlcuding default for port & qcmd here as well, just to simplify the launch bash script a little? (Unless there's a reason to do it there instead, of course)

Reply:
Can you have another look at this comment?